core/arch/arm/kernel/secure_partition.c

```diff
@@ -570,6 +570,55 @@ static TEE_Result handle_fdt_dev_regions(struct sp_ctx *ctx, void *fdt)
 	return TEE_SUCCESS;
 }
 
+static TEE_Result read_manifest_endpoint_id(struct sp_session *s,
+					    const void *fdt)
+{
+	uint32_t endpoint_id = 0;
+
+	/* Get the endpoint id defined in the manifest */
+	if (!sp_dt_get_u32(fdt, 0, "id", &endpoint_id)) {
+		struct sp_session *a_session = sp_get_session(endpoint_id);
+
+		if (endpoint_id <= SPMC_ENDPOINT_ID)
+			return TEE_ERROR_BAD_FORMAT;
+
+		/*
+		 * Switch the id's If the id was already assigned to a different
+		 * SP.
```

Comment:
This looks a bit strange despite it should works. Case where the manifest embeds nodes with identical  ID's would hit the panic in `check_endpoint_ids()`. 

By the way, prefer s/id/ID/g in inline comments.

Reply:
> This looks a bit strange despite it should works. Case where the manifest embeds nodes with identical ID's would hit the panic in `check_endpoint_ids()`.
We don't know the order in how the SPs will be processed. So a SP could define a ID in the manifest that is already taken automatically by another SP. This should work. We do however need to catch if 2 SPs define the same endpoint ID in their manifest file.
We could check the manifest file of the SPs we are getting the endpoint ID from. But the manifest file is currently not stored to the SP session or context. 
Any idea how to do it better?
> 
> By the way, prefer s/id/ID/g in inline comments.
ack

Reply:
How come we have this mix of how an `endpoint_id` is assigned? There seems to be one automatic and then this. I think this should be explained somewhere to make the code easier to read.

Reply:
I added some comment in the code.